src/main/java/it/pagopa/pn/api/dto/InputSearchNotificationDto.java

```diff
@@ -0,0 +1,124 @@
+package it.pagopa.pn.api.dto;
+
+import it.pagopa.pn.api.dto.notification.status.NotificationStatus;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.ToString;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Positive;
+import java.time.Instant;
+
+@EqualsAndHashCode
+@ToString
+@Getter
+public class InputSearchNotificationDto {
+    @NotEmpty
+    private final String senderReceiverId;
+    
+    @NotNull
+    private Instant startDate;
+    
+    @NotNull
+    private final Instant endDate;
+    
+    private final String filterId;
+    
+    private NotificationStatus status;
+    
+    private final String subjectRegExp;
+
+    @Positive
+    @NotNull
+    private final Integer size;
+    
+    private final String nextPagesKey;
+    
+    private final boolean bySender;
+    
+    public void setStartDate(Instant startDate){
+        this.startDate = startDate;
+    }
+    
+    public void setStatus(NotificationStatus status){
+        this.status = status;
+    }
+    
+    private InputSearchNotificationDto(Builder searchDtoBuilder){
+        this.senderReceiverId = searchDtoBuilder.senderReceiverId;
+        this.startDate = searchDtoBuilder.startDate;
+        this.endDate = searchDtoBuilder.endDate;
+        this.filterId = searchDtoBuilder.filterId;
+        this.status = searchDtoBuilder.status;
+        this.subjectRegExp = searchDtoBuilder.subjectRegExp;
+        this.size = searchDtoBuilder.size;
+        this.nextPagesKey = searchDtoBuilder.nextPagesKey;
+        this.bySender = searchDtoBuilder.bySender;
+    }
```

Comment:
Prima di finire il costruttore va invocata la validazione

Reply:
La validazione è stata portata fuori dal builder perchè non è possibile in model lanciare una PnValidationException